### ISSUE TITLE
TemplatePassthroughManagerCopyError when building Prod

### ISSUE THREAD

**Issue by mocsa** [dendronhq/dendron#775]
## Please select the area your Bug applys to. (Multiple selections are Possible)

- [ ] Onboard. Initial vault setup and import/export pods 
- [ ] Create. Note creation, lookup, snippets and templates
- [ ] Retrieve. Backlinks, references, graph view
- [ ] Structure. Refactoring, multi-vault and schemas
- [X] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Describe the bug

I cannot build a production site because of a build error.

Local preview works but even there a CSS file is missing and the page layout is unusable because I see huge icons.

## To Reproduce

1. Initialize a fresh workspace from VSCode.
2. In the workspace, issue the command: `npx dendron buildSite --stage prod`
3. I get a long list of errors which are shown in the log below.

## Expected behavior

I expect to have a working Prod site.

## Desktop (please complete the following information)

- OS: Win 10
- Browser Chrome
- Version 86
- Node Version v12.16.2

## Dendron Log file

```
{"level":30,"time":1622453406567,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByConfig","config":{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"useNoteTitleForLink":true,"noAutoCreateOnDefinition":true,"noLegacyNoteRef":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron","description":"Personal knowledge space","duplicateNoteBehavior":{"action":"useVault","payload":["vault"]},"siteUrl":"192.168.0.50","siteNotesDir":"notes","siteFaviconPath":"favicon.ico","gh_edit_link":true,"gh_edit_link_text":"Edit this page on GitHub","gh_edit_branch":"master","gh_root":"docs/","gh_edit_view_mode":"edit","writeStubs":true,"siteIndex":"root"},"initializeRemoteVaults":true}}
{"level":30,"time":1622453406569,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy","domain":"root","config":{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"useNoteTitleForLink":true,"noAutoCreateOnDefinition":true,"noLegacyNoteRef":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron","description":"Personal knowledge space","duplicateNoteBehavior":{"action":"useVault","payload":["vault"]},"siteUrl":"192.168.0.50","siteNotesDir":"notes","siteFaviconPath":"favicon.ico","gh_edit_link":true,"gh_edit_link_text":"Edit this page on GitHub","gh_edit_branch":"master","gh_root":"docs/","gh_edit_view_mode":"edit","writeStubs":true,"siteIndex":"root"},"initializeRemoteVaults":true}}
{"level":30,"time":1622453406570,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy:candidates","domain":"root","hConfig":{"publishByDefault":true,"noindexByDefault":false,"customFrontmatter":[]},"notes":["9379a7cd-12d1-4e17-9b4d-3ae9a82539dc"]}
{"level":30,"time":1622453406570,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy","fname":"root","domainNote":{"fname":"root","id":"9379a7cd-12d1-4e17-9b4d-3ae9a82539dc","children":[],"vault":{"fsPath":"vault"},"parent":null}}
copying assets from... { fsPath: 'vault' }
removing existing assets
{"level":30,"time":1622453406705,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByConfig","domains":["root"]}
{"level":30,"time":1622453406800,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByConfig","config":{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"useNoteTitleForLink":true,"noAutoCreateOnDefinition":true,"noLegacyNoteRef":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron","description":"Personal knowledge space","duplicateNoteBehavior":{"action":"useVault","payload":["vault"]},"siteUrl":"192.168.0.50","siteNotesDir":"notes","siteFaviconPath":"favicon.ico","gh_edit_link":true,"gh_edit_link_text":"Edit this page on GitHub","gh_edit_branch":"master","gh_root":"docs/","gh_edit_view_mode":"edit","writeStubs":true,"siteIndex":"root"},"initializeRemoteVaults":true}}
{"level":30,"time":1622453406801,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy","domain":"root","config":{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"useNoteTitleForLink":true,"noAutoCreateOnDefinition":true,"noLegacyNoteRef":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron","description":"Personal knowledge space","duplicateNoteBehavior":{"action":"useVault","payload":["vault"]},"siteUrl":"192.168.0.50","siteNotesDir":"notes","siteFaviconPath":"favicon.ico","gh_edit_link":true,"gh_edit_link_text":"Edit this page on GitHub","gh_edit_branch":"master","gh_root":"docs/","gh_edit_view_mode":"edit","writeStubs":true,"siteIndex":"root"},"initializeRemoteVaults":true}}
{"level":30,"time":1622453406802,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy:candidates","domain":"root","hConfig":{"publishByDefault":true,"noindexByDefault":false,"customFrontmatter":[]},"notes":["9379a7cd-12d1-4e17-9b4d-3ae9a82539dc"]}
{"level":30,"time":1622453406803,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy","fname":"root","domainNote":{"fname":"root","id":"9379a7cd-12d1-4e17-9b4d-3ae9a82539dc","children":[],"vault":{"fsPath":"vault"},"parent":null}}
{"level":30,"time":1622453406803,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByConfig","domains":["root"]}
{"level":30,"time":1622453406806,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByConfig","config":{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"useNoteTitleForLink":true,"noAutoCreateOnDefinition":true,"noLegacyNoteRef":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron","description":"Personal knowledge space","duplicateNoteBehavior":{"action":"useVault","payload":["vault"]},"siteUrl":"192.168.0.50","siteNotesDir":"notes","siteFaviconPath":"favicon.ico","gh_edit_link":true,"gh_edit_link_text":"Edit this page on GitHub","gh_edit_branch":"master","gh_root":"docs/","gh_edit_view_mode":"edit","writeStubs":true,"siteIndex":"root"},"initializeRemoteVaults":true}}
{"level":30,"time":1622453406807,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy","domain":"root","config":{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"useNoteTitleForLink":true,"noAutoCreateOnDefinition":true,"noLegacyNoteRef":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron","description":"Personal knowledge space","duplicateNoteBehavior":{"action":"useVault","payload":["vault"]},"siteUrl":"192.168.0.50","siteNotesDir":"notes","siteFaviconPath":"favicon.ico","gh_edit_link":true,"gh_edit_link_text":"Edit this page on GitHub","gh_edit_branch":"master","gh_root":"docs/","gh_edit_view_mode":"edit","writeStubs":true,"siteIndex":"root"},"initializeRemoteVaults":true}}
{"level":30,"time":1622453406808,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy:candidates","domain":"root","hConfig":{"publishByDefault":true,"noindexByDefault":false,"customFrontmatter":[]},"notes":["9379a7cd-12d1-4e17-9b4d-3ae9a82539dc"]}
{"level":30,"time":1622453406808,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByHiearchy","fname":"root","domainNote":{"fname":"root","id":"9379a7cd-12d1-4e17-9b4d-3ae9a82539dc","children":[],"vault":{"fsPath":"vault"},"parent":null}}
{"level":30,"time":1622453406809,"pid":17672,"hostname":"mydesktop","name":"logger","ctx":"filterByConfig","domains":["root"]}
Error with passthrough copy: (more in DEBUG output)
> Having trouble copying './assets/css/jquery.feedback_me.css'

`TemplatePassthroughManagerCopyError` was thrown
> Destination is not in the site output directory. Check your passthrough paths.

`TemplatePassthroughError` was thrown:
    TemplatePassthroughError: Destination is not in the site output directory. Check your passthrough paths.
        at TemplatePassthrough.copy (H:\FTPSync\server_folder\myworkspace\node_modules\@11ty\eleventy\src\TemplatePassthrough.js:93:7) 
        at TemplatePassthrough.write (H:\FTPSync\server_folder\myworkspace\node_modules\@11ty\eleventy\src\TemplatePassthrough.js:113:21)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
        at async Promise.all (index 0)
(node:17672) UnhandledPromiseRejectionWarning: TemplateWriterWriteError: Having trouble copying
    at H:\FTPSync\server_folder\myworkspace\node_modules\@11ty\eleventy\src\TemplateWriter.js:194:9
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:17672) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function 
without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)       
(node:17672) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Writing H:/FTPSync/server_folder/myworkspace/docs/CHANGELOG/index.html from ./CHANGELOG.md.
Writing H:/FTPSync/server_folder/myworkspace/docs/assets/js/just-the-docs.js from ./raw-assets/js/just-the-docs.js.liquid.
Writing H:/FTPSync/server_folder/myworkspace/docs/raw-assets/sass/vendor/normalize.scss/README/index.html from ./raw-assets/sass/vendor/normalize.scss/README.md.
Error writing templates: (more in DEBUG output)
> Having trouble copying

`TemplateWriterWriteError` was thrown
> Having trouble copying './assets/css/jquery.feedback_me.css'

`TemplatePassthroughManagerCopyError` was thrown
> Destination is not in the site output directory. Check your passthrough paths.

`TemplatePassthroughError` was thrown:
    TemplatePassthroughError: Destination is not in the site output directory. Check your passthrough paths.
        at TemplatePassthrough.copy (H:\FTPSync\server_folder\myworkspace\node_modules\@11ty\eleventy\src\TemplatePassthrough.js:93:7) 
        at TemplatePassthrough.write (H:\FTPSync\server_folder\myworkspace\node_modules\@11ty\eleventy\src\TemplatePassthrough.js:113:21)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
        at async Promise.all (index 0)
Problem writing Eleventy templates: (more in DEBUG output)
> Having trouble copying

`TemplateWriterWriteError` was thrown
> Having trouble copying './assets/css/jquery.feedback_me.css'

`TemplatePassthroughManagerCopyError` was thrown
> Destination is not in the site output directory. Check your passthrough paths.

`TemplatePassthroughError` was thrown:
    TemplatePassthroughError: Destination is not in the site output directory. Check your passthrough paths.
        at TemplatePassthrough.copy (H:\FTPSync\server_folder\myworkspace\node_modules\@11ty\eleventy\src\TemplatePassthrough.js:93:7) 
        at TemplatePassthrough.write (H:\FTPSync\server_folder\myworkspace\node_modules\@11ty\eleventy\src\TemplatePassthrough.js:113:21)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
        at async Promise.all (index 0)
Wrote 0 files in 0.34 seconds (v0.11.1)
dendron-cli.js buildSite

build notes for publication using 11ty

Opciók:
  --version         Verziószám megjelenítése                           [boolean]
  --help            Súgo megjelenítése                                 [boolean]
  --wsRoot          location of workspace
  --vault           name of vault
  --quiet           don't print output to stdout
  --enginePort      If set, connect to to running engine. If not set, create new
                    instance of Dendron Engine
  --useLocalEngine  If set, use in memory engine instead of connecting to a
                    server                                             [boolean]
  --serve           serve over local http server
                                              [boolean] [alapértelmezett: false]
  --stage           serve over local http server
                           [lehetőségek: "dev", "prod"] [alapértelmezett: "dev"]
  --output          if set, override output from config.yml             [szöveg]
  --custom11tyPath  if set, path to custom 11ty installation            [szöveg]

Error: H:\FTPSync\server_folder\myworkspace\docs\raw-assets\sass\just-the-docs-default.scss: no such file or directory
    at Object._newRenderError (H:\FTPSync\server_folder\myworkspace\node_modules\sass\sass.dart.js:13190:19)
    at StaticClosure._renderSync (H:\FTPSync\server_folder\myworkspace\node_modules\sass\sass.dart.js:13000:18)
    at Object.Primitives_applyFunction (H:\FTPSync\server_folder\myworkspace\node_modules\sass\sass.dart.js:1130:30)
    at Object.Function_apply (H:\FTPSync\server_folder\myworkspace\node_modules\sass\sass.dart.js:5897:16)
    at _callDartFunctionFast (H:\FTPSync\server_folder\myworkspace\node_modules\sass\sass.dart.js:7599:16)
    at Object.renderSync (H:\FTPSync\server_folder\myworkspace\node_modules\sass\sass.dart.js:7577:18)
    at buildSass (H:\FTPSync\server_folder\myworkspace\node_modules\@dendronhq\dendron-cli\node_modules\@dendronhq\dendron-11ty\bin\build-styles.js:12:22)
    at BuildSiteV2CLICommand.execute (H:\FTPSync\server_folder\myworkspace\node_modules\@dendronhq\dendron-cli\lib\src\commands\build-site-v2.js:121:28) {
  formatted: 'Error: H:\\FTPSync\\server_folder\\myworkspace\\docs\\raw-assets\\sass\\just-the-docs-default.scss: no such file or directory',
  status: 3
}
```

**Comment by timDeHof** [comment]
I'm getting the same error when using the command for local preview. 
command:npm dendron buildSite --stage dev --serve

**Comment by kevinslin** [comment]
can you check the version of dendron-cli and dendron-11ty? 

```
 npm info @dendronhq/dendron-cli 
 npm info @dendronhq/dendron-11ty
```

they should be 0.44 and 1.44 respectively

**Comment by mocsa** [comment]
Below is my output, executed directly in the VSCode terminal and in my workspace folder.

```
PS H:\FTPSync\server_folder\myworkspace> npm info @dendronhq/dendron-11ty

@dendronhq/dendron-11ty@1.44.0 | AGPLv3 | deps: 20 | versions: 104
dendron-11ty is a port of [dendron-jekyll](https://github.com/dendronhq/dendron-jekyll) from jekyll to eleventy.

dist
.tarball: https://registry.npmjs.org/@dendronhq/dendron-11ty/-/dendron-11ty-1.44.0.tgz
.shasum: 9ded7d52e37e4b02d2eb5f1a2ba7deca74674e6e
.integrity: sha512-BjXc0/0eBR96/Iyg2cwE4lEaCEATQZ/IkI3MtqcTGhvYt7mpFbM/KNOiEby9qtxMGj4VuKmJIUkEw2YN6VV6Sg==
.unpackedSize: 859.2 kB

dependencies:
@11ty/eleventy-plugin-syntaxhighlight: ^3.0.4 eleventy-xml-plugin: ^0.1.0                   rehype-raw: ^5.0.0
@11ty/eleventy: ^0.11.1                       fs-extra: ^9.0.1                              rehype-stringify: ^8.0.0
@dendronhq/common-all: ^0.44.0                liquidjs: ^9.16.1                             remark-rehype: ^8.0.0
@dendronhq/common-server: ^0.44.0             lodash: ^4.17.20                              remark: 12.0.1
@dendronhq/eleventy-plugin-seo: ^0.5.1        luxon: ^1.25.0                                request: ^2.88.2
@dendronhq/engine-server: ^0.44.0             markdown-it-task-checkbox: ^1.0.6             sass: ^1.32.7
@fec/eleventy-plugin-remark: ^2.1.0           markdown-it: ^12.0.3

maintainers:
- kevins8 <kevinslin8@gmail.com>

dist-tags:
latest: 1.44.0

published yesterday by kevins8 <kevinslin8@gmail.com>
PS H:\FTPSync\server_folder\myworkspace> npm info @dendronhq/dendron-cli  

@dendronhq/dendron-cli@0.44.0 | GPLv3 | deps: 12 | versions: 354
dendron-cli
https://github.com/dendronhq/dendron#readme

bin: dendron-cli, dendron

dist
.tarball: https://registry.npmjs.org/@dendronhq/dendron-cli/-/dendron-cli-0.44.0.tgz
.shasum: b08f85746d553a58e3ffc332fc94ea1c5f25b7e7
.integrity: sha512-6yJrH9XMHJDWHGew7GRDUz54IwPnulpuAysSGHk5tUwCOyOXNa4pQv8BUdbW+D0sxLhNspnOMtdunRqgeGricA==
.unpackedSize: 253.0 kB

dependencies:
@dendronhq/api-server: ^0.44.0    @dendronhq/dendron-11ty: ^0.44.0  @jcoreio/async-throttle: ^1.3.2   lodash: ^4.17.19
@dendronhq/common-all: ^0.44.0    @dendronhq/engine-server: ^0.44.0 fs-extra: ^9.0.1                  rsync: ^0.6.1
@dendronhq/common-server: ^0.44.0 @dendronhq/pods-core: ^0.44.0     gray-matter: ^4.0.2               yargs: ^15.4.1

maintainers:
- kevins8 <kevinslin8@gmail.com>

dist-tags:
latest: 0.44.0

published yesterday by kevins8 <kevinslin8@gmail.com>
```

But when I executed `npm list --depth=0` I saw old versions, like 1.42 and 0.43. Therefore I deleted the node_modules folder in my workspace and installed cli and 11ty from scratch (like in Getting Started->Publish). Then I saw the versions you mentioned after `npm list`. Therefore, I tried `npx dendron buildSite --stage prod` but I still got the same errors.

**Comment by cameronleeperlm** [comment]
I'm getting the same error when following preview site instructions: npm dendron buildSite --stage dev --serve

```
npm info @dendronhq/dendron-cli 

@dendronhq/dendron-cli@0.44.1 | GPLv3 | deps: 12 | versions: 356
dendron-cli
https://github.com/dendronhq/dendron#readme

bin: dendron-cli, dendron

dist
.tarball: https://registry.npmjs.org/@dendronhq/dendron-cli/-/dendron-cli-0.44.1.tgz
.shasum: ea85f151e7d2c2380834c5872405d9a29a68da65
.integrity: sha512-MYZsHUOH6Ga1frkCodJoCkCt+DnfxsPwGdNt4H4F5KEnj/KcCN+lrjcu2q5HAvOHNeNN2HQQypN68ecFPqAAeQ==
.unpackedSize: 255.7 kB

@dendronhq/api-server: ^0.44.1    @dendronhq/dendron-11ty: ^0.44.1  @jcoreio/async-throttle: ^1.3.2   lodash: ^4.17.19
@dendronhq/common-all: ^0.44.1    @dendronhq/engine-server: ^0.44.1 fs-extra: ^9.0.1                  rsync: ^0.6.1
@dendronhq/common-server: ^0.44.1 @dendronhq/pods-core: ^0.44.1     gray-matter: ^4.0.2               yargs: ^15.4.1

maintainers:
- kevins8 <kevinslin8@gmail.com>

dist-tags:
latest: 0.44.1

published 2 days ago by kevins8 <kevinslin8@gmail.com>
```
```
> npm info @dendronhq/dendron-11ty

@dendronhq/dendron-11ty@0.44.1 | AGPLv3 | deps: 20 | versions: 107
dendron-11ty is a port of [dendron-jekyll](https://github.com/dendronhq/dendron-jekyll) from jekyll to eleventy.

dist
.tarball: https://registry.npmjs.org/@dendronhq/dendron-11ty/-/dendron-11ty-0.44.1.tgz
.shasum: 50e84b5a42018f44ef8a2e548021b9c1a2b92d18
.integrity: sha512-OTN9SAwn7Kj0j9xgt9SvcYM/0EIsrINk/dYyLn954adtkwaB1j3ODoHp3OU4PU8zaWiYcu2CHTJgr69momqSsA==
.unpackedSize: 860.8 kB

dependencies:
@11ty/eleventy-plugin-syntaxhighlight: ^3.0.4 eleventy-xml-plugin: ^0.1.0                   rehype-raw: ^5.0.0
@11ty/eleventy: ^0.11.1                       fs-extra: ^9.0.1                              rehype-stringify: ^8.0.0
@dendronhq/common-all: ^0.44.1                liquidjs: ^9.16.1                             remark-rehype: ^8.0.0
@dendronhq/common-server: ^0.44.1             lodash: ^4.17.20                              remark: 12.0.1
@dendronhq/eleventy-plugin-seo: ^0.5.1        luxon: ^1.25.0                                request: ^2.88.2
@dendronhq/engine-server: ^0.44.1             markdown-it-task-checkbox: ^1.0.6             sass: ^1.32.7
@fec/eleventy-plugin-remark: ^2.1.0           markdown-it: ^12.0.3

maintainers:
- kevins8 <kevinslin8@gmail.com>

dist-tags:
latest: 0.44.1

published 2 days ago by kevins8 <kevinslin8@gmail.com>
```

**Comment by flammehawk** [comment]
@kevinslin 
> can you check the version of dendron-cli and dendron-11ty?
> 
> ```
>  npm info @dendronhq/dendron-cli 
>  npm info @dendronhq/dendron-11ty
> ```
> 
> they should be 0.44 and 1.44 respectively

So what is the actual version of ` @dendronhq/dendron-11ty ` ? 
is it 1.44 or is it 0.44 ? 

If you look at the [versions here] (https://npm.io/package/@dendronhq/dendron-11ty) then you may notice that it is completely confusing it feels like it jumps from 1.xx versions back to 0.xx Versions.

**Comment by kevinslin** [comment]
@flammehawk it should be 1.44. the reason for the confusing version numbers is because 11ty used to be a standalone package, not part of the monorepo. we've added it to the monorepo later which uses synced versions (all packages have the same version). currently, this is 0.X. because 11ty was published as 1.X and we can't unpublish a package in npm, we are doing a separate 1.X release for dendron-11ty on every release

**Comment by kevinslin** [comment]
this issue should now be fixed in 0.45.1 version of the dendron-cli. 

you can install it using
```
yarn add @dendronhq/dendron-cli@0.45.1
```

note that if you use `site build` from the plugin, it will still fail since its using the old version. this fix will roll out to the plugin for next mondays release.

**Comment by mocsa** [comment]
I reinstalled dendron-cli with

```
npm install @dendronhq/dendron-cli@latest
```

and I can confirm that I no longer experience the problem. Thank you.

**Comment by kevinslin** [comment]
this is fixed with 0.46 release
